### ISSUE TITLE
Bound function has no access to this.refs

### ISSUE THREAD

**Issue by coodoo** [facebook/react#7373]
Seemed when a function is bound to the `component` class, it can *no longer* access `this.refs`, see following simple case, where `bar()` works but `foo()` didn't, I would expect the two should behave the same, any idea?

```javascript
import React, { Component, PropTypes } from 'react'

export default class Sample extends Component {

	constructor( props ) {
		super( props )
	}

	componentDidUpdate() {
		console.log( 'cDU: ', this.refs ) // tada
		this.foo()
		this.bar()
	}

	render() {
		return <div ref='tada'>bar</div>
	}

	// doesn't work
	foo = evt => {
		console.log( 'ref: ', this.refs ) // null
	}

	// works
	bar(evt) {
		console.log( 'ref: ', this.refs ) // tada
	}
}
```

**Comment by syranide** [comment]
You are binding it to the global `this` and then assigning it to the property I would assume.

**Comment by coodoo** [comment]
Confirmed fixed in `v15.3.0`, great jobs!
